### DISCUSSION TITLE
Proposal: null conditional + lambda expression

### DISCUSSION THREAD

**Discussion by murtazadhari** [dotnet/csharplang#1676] [Language Ideas]
Hi, It would be good to have something like 
`home?=> new PlayerSummary(home) `
for scenario 
`var h = home != null ? new HomeSummary(home) : null`

so instead of writing ternary operator of if/else we could just write null conditional + return lambda expression. 

`var h = home?=> new HomeSummary(home).`

Cheers
Ali, Murtaza

**Comment by stakx** [comment]
This could be covered by e.g. #1661.

**Comment by DavidArno** [comment]
@stakx,

The idea behind the likes of #1661 is that a ArgumentNullException` is thrown, which is orthogonal to this proposal.

Having said that, a proposal for shortcut syntax for only reassigning a value if it's not null is the complete opposite of "would be good to have". I just don't get why such logic would ever exist, much less want a shorthand way of expressing it.

**Comment by Joe4evr** [comment]
> a proposal for shortcut syntax for only reassigning a value if it's not null is the complete opposite of "would be good to have"

Not to mention that OP's request can already be achieved in a different manner.
```cs
//Add method on your 'Home' type:
public HomeSummary CreateSummary() => new HomeSummary(this);

//Call like so
var h = home?.CreateSummary();
```
Problem solved, no new syntax necessary.

**Comment by stakx** [comment]
@DavidArno: #1661 is not about throwing an exception as you claim, but about skipping method execution if any one of the marked arguments is `null`. It says so in the very first sentence and in the example.

**Comment by jnm2** [comment]
See https://github.com/dotnet/csharplang/issues/433, as well as another duplicate, https://github.com/dotnet/csharplang/issues/842.
